### PR TITLE
Fix/when virustotal API reaches rate limit should not crash.

### PR DESCRIPTION
**Pull request by oussamaessaji**

### CODE REVIEW

File: agent/virustotal.py [hunk 1]
```diff
@@ -55,7 +58,10 @@ def get_scans(response: dict[str, Any]) -> dict[str, Any] | None:
     Raises:
         VirusTotalApiError: In case the API request encountered problems.
     """
-    if response["response_code"] == 0 or "results" not in response:
+    if response.get("response_code") == 204 and response.get("error") is not None:
+        logger.error("Exceeded the virustotal API rate limit.")
+        raise VirusTotalApiError()
```
Comment (3asm) [inline agent/virustotal.py:63]: And we crash???

**Review by 3asm — CHANGES_REQUESTED** [review]

File: tests/test_agent.py [hunk 2]
```diff
@@ -228,3 +228,29 @@ def testVirusTotalAgent_whenApisReceived_virusTotalApiReturnsValidResponse(
...
+            "error": "You exceeded the public API request rate limit (4 requests of any nature per minute)",
+            "response_code": 204,
+        }
+
+    mocker.patch(
+        "virus_total_apis.PublicApi.get_file_report",
+        side_effect=virustotal_invalid_response,
+    )
+
+    with pytest.raises(virustotal.VirusTotalApiError):
```
Comment (mohsinenar) [inline tests/test_agent.py:255]: It's not supposed to crash.

**Review by mohsinenar — CHANGES_REQUESTED** [review]

File: agent/virustotal.py (hunk 1 above)
Comment (oussamaessaji) [reply agent/virustotal.py:63]: We do catch it outside.

File: tests/test_agent.py (hunk 2 above)
Comment (oussamaessaji) [reply tests/test_agent.py:255]: get scans will crash but this is a custom exception that we catch outside

**Review by amine3 — APPROVED** [review]

File: tests/test_agent.py (hunk 2 above)
Comment (3asm) [reply tests/test_agent.py:255]: This is process crashing, where is it caught!!

**Review by 3asm — CHANGES_REQUESTED** [review]

File: agent/virustotal.py (hunk 1 above)
Comment (amine3) [reply agent/virustotal.py:63]: You should remove the raise exception in _process_response when you catch VirusTotalApiError

**Review by amine3 — CHANGES_REQUESTED** [review]

**Review by amine3 — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**